src/main/java/de/tum/in/www1/artemis/web/rest/SubmissionResource.java

```diff
@@ -104,9 +129,41 @@ public SubmissionResource(SubmissionRepository submissionRepository) {
      * @return the ResponseEntity with status 200 (OK)
      */
     @DeleteMapping("/submissions/{id}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<Void> deleteSubmission(@PathVariable Long id) {
         log.debug("REST request to delete Submission : {}", id);
+
+        Optional<Submission> submission = submissionRepository.findById(id);
+
+        if (!submission.isPresent()) {
+            log.error("Submission with id: " + id + " cannot be deleted");
+            return ResponseEntity.notFound().build();
+        }
+
+        checkAccessPermissionAtInstructor(submission.get());
+
+        if (submission.get().getResult() != null) {
+            resultRepository.delete(submission.get().getResult());
+        }
         submissionRepository.deleteById(id);
+
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, id.toString())).build();
     }
+
+    private void checkAccessPermissionAtInstructor(Submission submission) {
+        Course course = findCourseFromSubmission(submission);
+        User user = userService.getUserWithGroupsAndAuthorities();
+
+        if (!authCheckService.isAtLeastInstructorInCourse(course, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
```

Comment:
We normally you don't use this exception for returning a response, instead you could throw "IllegalAccess" and catch that in the view method.

Besides that: This should probably be in the AuthorizationCheckService, not here.